### PR TITLE
Implement suggestion for edit composer

### PR DESCRIPTION
**Pull request by dukenv0307**
### Fixed Issues

$ https://github.com/Expensify/App/issues/34442
PROPOSAL: https://github.com/Expensify/App/issues/34442#issuecomment-1891334360

### Tests

1. Open a long-history report
2. Send a message
3. Hover or long-press over the message and select `Edit message`
4. Typing `@` and verify that the mention suggestion is opened
5. Typing a part of an emoji name like `:sm` and verify that the emoji suggestion is opened
6. Keep the suggestion open and scroll the report screen
7. Verify that the suggestion is closed
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above
### QA Steps

1. Open a long-history report
2. Send a message
3. Hover or long-press over the message and select `Edit message`
4. Typing `@` and verify that the mention suggestion is opened
5. Typing a part of an emoji name like `:sm` and verify that the emoji suggestion is opened
6. Keep the suggestion open and scroll the report screen
7. Verify that the suggestion is closed
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/129500732/2a56ed3a-f5c7-4d94-b0e0-1c47f80c89d8

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/129500732/ccdf3d04-1c38-4594-85ab-2e917049d1fb

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/129500732/57b11a49-274e-46a5-aaa0-1febfddefc70

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/129500732/18eb167a-78a1-4874-a165-076d11935845

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/129500732/9a01956c-7016-495b-ade7-de624a8086f7

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/129500732/7592b2cf-18bc-4229-a2e0-e70ec66f3d5f

</details>

### CODE REVIEW

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Review by eVoloshchak — COMMENTED** [review]
> Suggestions container for edit composer has a smaller width than for the default composer (notice the blue avatar is partially visible)

The key here is to have two avatars alongside

1. Press FAB -> Request Money
2. Enter an amount, press next
3. Add two participants to Split, press `Split..` to confirm
4. Click on split to open the details, select one of the participants, select `Message participant`
5. Open emoji suggestion modal for normal and edit composer to observe the difference

https://github.com/Expensify/App/assets/9059945/832b360a-da77-400a-91ff-c740eb70dbdd

**Comment by eVoloshchak** [comment]
> I'm closing suggestion for edit composer since the suggestion has a fixed position when it's opened, it will not move when we scroll the report screen

Is there a way we could make it scroll with the report screen?
This doesn't look like the desired behavior

https://github.com/Expensify/App/assets/9059945/8a7b137f-69aa-4738-af3e-6b89824b30fc

**Review by eVoloshchak — COMMENTED** [review]
There are a couple of bugs:

1. Web/Desktop, edit composer isn't closed when clicking outside of it

https://github.com/Expensify/App/assets/9059945/fec24890-2a6c-4c61-b4fe-b4e8846843d2

2. Composer isn't closed when you navigate to a different chat

https://github.com/Expensify/App/assets/9059945/bef80121-2998-4b77-bc1c-817022ec4790

**Comment by eVoloshchak** [comment]
Suggestions list sits 2px on top of edit composer
Screenshot 2024-02-28 at 21 57 14
It should have a margin, same as the regular composer
Screenshot 2024-02-28 at 21 56 56

**Review by eVoloshchak — COMMENTED** [review]
> @eVoloshchak what does that mean? because the edit composer has padding top-bottom 8px and has bgcolor so I think the suggestion should display above 8px which is fit with the padding of the edit composer.

@dukenv0307, notice the suggestions window composer overlapping with edit composer's grey background (just a couple of px it seems) 
Screenshot 2024-03-12 at 19 33 49

Screenshot 2024-03-12 at 19 33 54

**Review by eVoloshchak — COMMENTED** [review]
Bug:

1. Start editing a message
2. Start typing an emoji name so suggestions modal is opened
3. Click on the normal composer (the global one)
4. Notice the global composer isn't focused
5. Notice emoji suggestions modal isn't closed


https://github.com/Expensify/App/assets/9059945/2e17294c-7859-486a-a7e5-a1a769bcfc3b

**Review by eVoloshchak — COMMENTED** [review]
>  Do you think we can change SUGGESTER_PADDING to 8 as well? I think that will be good because it just increases the space between the suggestion and the composer

We shouldn't change the way it currently looks for the global composer, I think we have to add a variable specifically for edit composer instead of changing `SUGGESTER_PADDING`

> Can't reproduce the bug above

Yeah, the steps turned out to be more specific
1. Navigate to any chat that has a thread with dome relies or a task in it
2. Send a message, start editing the message
3. Start typing an emoji name so suggestions modal is opened
4. Click on replies or a task (so you're navigated to a different chat)
5. Navigate back (click on back button)
6. Continue typing an emoji so suggestions modal is opened
7. Click on the global composer 


https://github.com/Expensify/App/assets/9059945/9d0ff8de-9f59-461f-8e4d-cf46daff779e

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 1]
```diff
@@ -363,6 +378,25 @@ function ReportActionItemMessageEdit(
...
+    }, [textInputRef]);
+
+    const measureContainer = useCallback(
+        (callback: MeasureInWindowOnSuccessCallback) => {
+            if (!containerRef.current) {
+                return;
+            }
+            containerRef.current.measureInWindow(callback);
+        },
+        // We added isComposerFullSize in dependencies so that when this value changes, we recalculate the position of the popup
```
Comment (eVoloshchak) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:395]: This comment isn't needed I think, we don't use `isComposerFullSize` here

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 2]
```diff
@@ -18,11 +19,13 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainer = () => {}
     const StyleUtils = useStyleUtils();
     const containerRef = React.useRef<HTMLDivElement>(null);
     const {windowHeight, windowWidth} = useWindowDimensions();
+    const suggestionContainerHeight = measureHeightOfSuggestionsContainer(props.suggestions.length, props.isSuggestionPickerLarge);
     const [{width, left, bottom}, setContainerState] = React.useState({
         width: 0,
         left: 0,
         bottom: 0,
     });
+    const [shouldBelowContainer, setShouldBelowContainer] = React.useState(false);
```
Comment (eVoloshchak) [inline src/components/AutoCompleteSuggestions/index.tsx:28]:
```suggestion
    const [shouldShowBelowContainer, setShouldShowBelowContainer] = React.useState(false);
```

File: src/components/AutoCompleteSuggestions/index.tsx [hunk 3]
```diff
@@ -41,13 +44,19 @@ function AutoCompleteSuggestions<TSuggestion>({measureParentContainer = () => {}
         if (!measureParentContainer) {
             return;
         }
-        measureParentContainer((x, y, w) => setContainerState({left: x, bottom: windowHeight - y, width: w}));
-    }, [measureParentContainer, windowHeight, windowWidth]);
+
+        measureParentContainer((x, y, w, h) => {
+            const currenBottom = y < suggestionContainerHeight ? windowHeight - y - suggestionContainerHeight - h : windowHeight - y;
```
Comment (eVoloshchak) [inline src/components/AutoCompleteSuggestions/index.tsx:49]:
```suggestion
            const currentBottom = y < suggestionsContainerHeight ? windowHeight - y - suggestionContainerHeight - h : windowHeight - y;
```

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
@dukenv0307, the code looks good, couldn't find any issues during the initial testing, awesome job!
Could you please merge the latest main and upload screen recordings for iOS/Android native?

**Comment by eVoloshchak** [comment]
Bug: 

1. Start editing a message
2. Start entering an emoji name so that the suggestions modal is shown (i.e. `:sm`)
3. While keeping the composer focused, scroll the chat window up or down so that the edit composer is hidden behind the header or global composer
4. Continue typing the emoji name 
5. Notice the emoji suggestions modal opens, chat is scrolled (so that edit composer is visible), which causes emoji suggestions modal to close immediately

This is kind of an edge case not many users will face, but still a valid issue IMO

https://github.com/Expensify/App/assets/9059945/0da1e03e-9f15-4ab4-99f7-8418804714dd

**Review by eVoloshchak — COMMENTED** [review]
> the users will not scroll to hide the edit composer if they want to edit

This can also happen if you resize the window or change the device orientation. Let's try to fix this, it's a minor bug, but still a bug

**Review by eVoloshchak — COMMENTED** [review]
> And I also updated to re-open the suggestion after the scroll had ended. Do you agree with that?

@dukenv0307, I don't think that's the correct behavior, it should be consistent with emoji picker
This also yields another bug when you manually scroll the chat

https://github.com/Expensify/App/assets/9059945/f86112e8-854d-4ad3-9492-7055ea01d039

File: src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx [hunk 4]
```diff
@@ -377,7 +378,7 @@ function ReportActionCompose({
                 {shouldShowReportRecipientLocalTime && hasReportRecipient && <ParticipantLocalTime participant={reportRecipient} />}
             </OfflineWithFeedback>
             <View style={isComposerFullSize ? styles.flex1 : {}}>
-                <PortalHost name="suggestions" />
+                <PortalHost name="suggestions_0" />
```
Comment (eVoloshchak) [inline src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx:381]: This ("suggestions_0" string) would get lost during a refactor, it's not clear why we have this specific name here. I propose we create a new key in CONST, use it here and then in `src/components/AutoCompleteSuggestions/index.native.tsx` we could do something along the lines of `<Portal hostName={ activeID ? "suggestions_${activeID}" : CONST.DEFAULT_COMPOSER_PORTAL_HOST_NAME}>`

**Review by eVoloshchak — CHANGES_REQUESTED** [review]
Other than the small change below this looks good from the UI and functionality standpoint!
Could you merge main and I'll finish up the checklist?

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by eVoloshchak — COMMENTED** [review]
@dukenv0307, could you include some test steps so the QA team could test this?

**Review by eVoloshchak — APPROVED** [review]
LGTM
Awesome work, @dukenv0307!

File: src/libs/SuggestionUtils.ts [hunk 5]
```diff
@@ -20,4 +20,24 @@ function hasEnoughSpaceForLargeSuggestionMenu(listHeight: number, composerHeight
     return availableHeight > menuHeight;
 }
 
-export {trimLeadingSpace, hasEnoughSpaceForLargeSuggestionMenu};
+const measureHeightOfSuggestionsContainer = (numRows: number, isSuggestionsPickerLarge: boolean): number => {
+    const borderAndPadding = CONST.AUTO_COMPLETE_SUGGESTER.SUGGESTER_PADDING + 2;
```
Comment (stitesExpensify) [inline src/libs/SuggestionUtils.ts:24]: why +2? This feels arbitrary without context. I think we should either have a const or a detailed comment

File: src/pages/home/report/ReportActionItemMessageEdit.tsx [hunk 6]
```diff
@@ -374,6 +390,24 @@ function ReportActionItemMessageEdit(
...
+    }, [textInputRef]);
+
+    const measureContainer = useCallback(
+        (callback: MeasureInWindowOnSuccessCallback) => {
+            if (!containerRef.current) {
+                return;
+            }
+            containerRef.current.measureInWindow(callback);
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (stitesExpensify) [inline src/pages/home/report/ReportActionItemMessageEdit.tsx:407]: Why is this necessary?

**Review by stitesExpensify — COMMENTED** [review]
Looks great! Super excited for this feature. Just 2 quick questions

**Review by stitesExpensify — APPROVED** [review]
LGTM! Excited for this feature

**Comment by Julesssss** [comment]
Hey @stitesExpensify @dukenv0307. I think [this Android bug](https://github.com/Expensify/App/issues/40854) is liked to these changes. I don't think it's serious enough to revert, but it would be great to get it fixed asap. Any thoughts?

**Comment by stitesExpensify** [comment]
@dukenv0307 looks like we had to revert this pr due to deploy blockers it spawned. Can you create a revert revert and fix those problems so we can try again please?

**Pull request merged**